### PR TITLE
Minor improvements for Windows compatibility

### PR DESCRIPTION
**Pull request by szotov**
Helps with developing smart contracts (not full compatibility)

### CODE REVIEW

File: nucypher/config/utils.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def check_config_tree(configuration_dir: str=None) -> bool:
 
 def check_config_runtime() -> bool:
     rules = (
-        (os.getuid() != 0, 'Cannot run as root user.'),
+        (os.name == 'nt' or os.getuid() != 0, 'Cannot run as root user.'),
```
Comment (michwill) [inline nucypher/config/utils.py]: Presumably, ok if `Administrator`?

File: tests/fixtures.py [hunk 2]
```diff
@@ -118,6 +118,7 @@ def tempfile_path():
     """
     User is responsible for closing the file given at the path.
     """
-    _, path = tempfile.mkstemp()
+    fd, path = tempfile.mkstemp()
     yield path
+    os.close(fd)
```
Comment (michwill) [inline tests/fixtures.py]: So, Windows doesn't complain that this file is opened elsewhere?

File: nucypher/config/utils.py (hunk 1 above)
Comment (szotov) [reply nucypher/config/utils.py]: For now it's not for real usage on Windows, so it's ok

File: tests/fixtures.py (hunk 2 above)
Comment (szotov) [reply tests/fixtures.py]: File opened in the same process using `with` statement, I'm not sure why it still open

**Comment by KPrasch** [comment]
This will need a rebase against master after merging #286

**Comment by tuxxy** [comment]
What problem is this exactly solving for Windows? I don't have anything to run it on.

**Comment by szotov** [comment]
These changes allow to run blockchain tests on Windows with a predefined environment. There are no tests for those lines, main goal is not to break existing code/tests

**Comment by michwill** [comment]
I guess, we should rebase it with recent master and merge?

**Pull request merged**